src/types/dto/tasks.dto.ts

```diff
@@ -0,0 +1,22 @@
+import { z } from 'zod'
+
+export const AssigneeTypeSchema = z.enum(['internalUser', 'client', 'company']).optional()
+export type AssigneeType = z.infer<typeof AssigneeTypeSchema>
+
+export const CreateTaskRequestSchema = z.object({
+  assigneeId: z.string().optional(),
+  assigneeType: AssigneeTypeSchema,
+  title: z.string(),
+  body: z.string().optional(),
+  workflowStateId: z.string().uuid(),
+})
+export type CreateTaskRequest = z.infer<typeof CreateTaskRequestSchema>
+
+export const UpdateTaskRequestSchema = z.object({
```

Comment:
This looks same as Create schema just with optional, in Zod, we can achieve this using partial

CreateTaskRequestSchema.partial()